### PR TITLE
Enforce integration test check on PRs

### PR DESCRIPTION
**Pull request by marcellp**
# Description
Enforces integration tests to run before merging. testing

# Details
#### URL to issue
https://biomage.atlassian.net/browse/BIOMAGE-1501

#### Link to staging deployment URL (or set N/A)
https://ui-marcellp-mandatory-e2e.scp-staging.biomage.net/

#### Integration test branch
fix/fix-tests

# Merge checklist
Your changes will be ready for merging after all of the steps below have been completed.

### Code updates
Have best practices and ongoing refactors being observed in this PR
- [ ] Migrated any selector / reducer used to the new format.

### Manual/unit testing
- [ ] Unit tests written **or** no unit tests required for change, e.g. documentation update.
- [ ] Tested changes using InfraMock locally **or** no tests required for change, e.g. Kubernetes chart updates.
- [ ] Validated that current unit tests for code work as expected and are sufficient for code coverage **or** no unit tests required for change, e.g. documentation update.

### Integration testing
End-to-end tests have to pass before the PR can be merged. **You must check the box below** to run these
tests on the latest commit on your PR branch. Without doing so, your PR **will not pass** the required
status checks for merging.

- [x] Started end-to-end tests on the latest commit.

### Documentation updates
- [ ] Relevant Github READMEs updated **or** no GitHub README updates required.
- [ ] Relevant Wiki pages created/updated **or** no Wiki updates required.

### Optional
- [ ] Photo of a cute animal attached to this PR.

### CODE REVIEW

File: pr_body.md
Comment (marcellp) [inline pr_body.md]: Remove this file.

File: .github/pull_request_template.md
Comment (marcellp) [inline .github/pull_request_template.md]: Make this "links to any PRs OR internal resources related to this PR" (e.g. slack messages).

rephrase it to make it sound better

**Comment by aerlaut** [comment]
Testing a comment

File: .github/pull_request_template.md
Comment (aerlaut) [reply .github/pull_request_template.md]: changed

**Comment by aerlaut** [comment]
New comment

**Comment by aerlaut** [comment]
New comment

**Comment by aerlaut** [comment]
Testing new trigger

**Review by marcellp — COMMENTED** [review]
sample comment

File: .github/pull_request_template.md
Comment (marcellp) [inline .github/pull_request_template.md:2]: testing trigger

**Comment by marcellp** [comment]
/e2e

asdasd

**Review by cosa65 — DISMISSED** [review]

**Review by cosa65 — DISMISSED** [review]

**Review by cosa65 — DISMISSED** [review]

**Review by cosa65 — DISMISSED** [review]

File: .github/pull_request_template.md
Comment (ivababukova) [inline .github/pull_request_template.md]: can you make this more generic, so that it applies not only for jira issues, but also for other issues (like github issues potentially in the future)?

File: .github/pull_request_template.md
Comment (ivababukova) [inline .github/pull_request_template.md]: what if we want to merge in a very simple change (like a change of README file)? Will then our pull request be blocked until we stage the change?

File: .github/pull_request_template.md
Comment (ivababukova) [inline .github/pull_request_template.md]: why adding `internal` in front of resources? Makes it ambiguous, better just say `resources`

File: .github/pull_request_template.md
Comment (ivababukova) [inline .github/pull_request_template.md]: why will it not pass?

File: .github/pull_request_template.md
Comment (ivababukova) [inline .github/pull_request_template.md]: this is confusing. My first thought when I read it was that this is the branch of the api the tests will be run against. Rephrase the sentence to make it clear that this is the integration tests branch

File: .github/pull_request_template.md
Comment (ivababukova) [inline .github/pull_request_template.md]: I would reorder these three so that the unit tests come one after the other and so that it is clear that one of the unit tests checks is about fixing existing tests and the other one is about writing new tests. You might consider merging them into 1 as well.

File: .github/pull_request_template.md
Comment (ivababukova) [inline .github/pull_request_template.md]: Call them `integration tests`, not `end-to-end` tests because it makes people think that you might be referring to some other tests

File: .github/pull_request_template.md
Comment (ivababukova) [inline .github/pull_request_template.md]: add a link to the biomage utils repo saying that people can stage from there

File: .github/pull_request_template.md
Comment (ivababukova) [inline .github/pull_request_template.md:72]: add a checkbox for unstaging after merging the PR

File: .github/workflows/pr_validate.yaml
Comment (ivababukova) [inline .github/workflows/pr_validate.yaml]: why do you need this?

Comment (ivababukova) [reply .github/workflows/pr_validate.yaml]: We cannot couple JIRA with out build system, because that prevents open source contributors to merge pull requests unless they are added to jira

File: .github/workflows/pr_validate.yaml
Comment (ivababukova) [inline .github/workflows/pr_validate.yaml]: call it integration tests

File: .github/workflows/pr_validate.yaml
Comment (ivababukova) [inline .github/workflows/pr_validate.yaml]: call it integration tests

File: .github/workflows/pr_validate.yaml
Comment (ivababukova) [inline .github/workflows/pr_validate.yaml:176]: call it integration tests

File: .github/workflows/pr_validate.yaml
Comment (marcellp) [reply .github/workflows/pr_validate.yaml:38]: As we discussed, this is not necessary, you can leave it unfilled by adding N/A. This way the system knows that you deliberately ignored it instead of forgetting to fill it in.

File: .github/workflows/pr_validate.yaml
Comment (marcellp) [reply .github/workflows/pr_validate.yaml:86]: Done.

File: .github/workflows/pr_validate.yaml
Comment (marcellp) [reply .github/workflows/pr_validate.yaml:87]: Done.

Comment (marcellp) [reply .github/workflows/pr_validate.yaml]: Done.

File: .github/workflows/pr_validate.yaml
Comment (marcellp) [reply .github/workflows/pr_validate.yaml]: That will not be a problem, you can replace the JIRA URL with N/A and it will pass. It's just to make sure you are deliberately not including one.

File: .github/pull_request_template.md
Comment (marcellp) [reply .github/pull_request_template.md]: Sure, done.

File: .github/pull_request_template.md
Comment (marcellp) [reply .github/pull_request_template.md]: Done.

File: .github/pull_request_template.md
Comment (marcellp) [reply .github/pull_request_template.md]: Done.

File: .github/pull_request_template.md
Comment (marcellp) [reply .github/pull_request_template.md:72]: Done.

File: .github/pull_request_template.md
Comment (marcellp) [reply .github/pull_request_template.md]: It says it there in the second line. Do you think it should be rephrased as it's ambiguous or was it just a mistake of not noticing it?

File: .github/pull_request_template.md
Comment (marcellp) [reply .github/pull_request_template.md]: Fixed.

File: .github/pull_request_template.md
Comment (marcellp) [reply .github/pull_request_template.md]: It will, I just copy-pasted it. Now changed.

File: .github/workflows/pr_validate.yaml
Comment (marcellp) [reply .github/workflows/pr_validate.yaml]: I did remove all references to JIRA though (outside of the actual validator obviously) so when we change to GItHub we can just replace the regex.

File: .github/pull_request_template.md
Comment (ivababukova) [inline .github/pull_request_template.md]: remove this hardcoded N/A?

File: .github/pull_request_template.md
Comment (ivababukova) [reply .github/pull_request_template.md]: It is a little ambiguous, maybe something like this is better:


```
Delete this comment and include the URL of the staging environment for this pull request, or write "N/A" and explain why a staging environment is not necessary.
Your pull request will not pass the required checks if this is not followed.
```

File: .github/workflows/ci.yaml
Comment (ivababukova) [inline .github/workflows/ci.yaml]: did you comment this out for debugging purposes?

File: .github/workflows/pr_validate.yaml
Comment (ivababukova) [inline .github/workflows/pr_validate.yaml]: did you comment this out for debugging purposes?

Comment (marcellp) [reply .github/workflows/pr_validate.yaml]: yes

File: .github/workflows/ci.yaml
Comment (marcellp) [reply .github/workflows/ci.yaml]: yup!

File: .github/pull_request_template.md
Comment (aerlaut) [reply .github/pull_request_template.md]: Removed. I agree it should fail by default.

File: .github/pull_request_template.md
Comment (aerlaut) [reply .github/pull_request_template.md]: added

File: .github/pull_request_template.md
Comment (aerlaut) [reply .github/pull_request_template.md]: Rephrased. What do you think?

File: .github/pull_request_template.md
Comment (aerlaut) [reply .github/pull_request_template.md]: Reworded

File: .github/pull_request_template.md
Comment (ivababukova) [inline .github/pull_request_template.md]: feels like this belongs to the `#### Link to staging deployment URL` section, not here under `#### URL to issue`

**Review by ivababukova — DISMISSED** [review]

File: .github/pull_request_template.md
Comment (aerlaut) [reply .github/pull_request_template.md]: moved

**Review by cosa65 — APPROVED** [review]

File: .github/workflows/pr_validate.yaml
Comment (aerlaut) [inline .github/workflows/pr_validate.yaml]: We have to run integration tests even if the box is not checked because a skipped job does not equal to a failed / cancelled one and we can not use it as a status check.

**Pull request merged**
